### ISSUE TITLE
Error in Parsing Cert-Mananger `Challenge` issue

### ISSUE THREAD

**Issue by hpsawant7887-lacer** [kubernetes/kube-state-metrics#1862]
**What happened**:
I am trying to create custom metrics from Cert-Manager Resources like Challenge, Certificate etc. 

I followed this document -> https://github.com/kubernetes/kube-state-metrics/blob/master/docs/customresourcestate-metrics.md

Currently I get an error when trying to parse, `Challenge` CRD

***Challenge resource***
```
apiVersion: acme.cert-manager.io/v1
kind: Challenge
metadata:
  creationTimestamp: "2022-10-19T21:21:58Z"
  finalizers:
  - finalizer.acme.cert-manager.io
  generation: 1
  name: demo2-hsawant-tzlw7-616897090-806614722
  namespace: default
  ownerReferences:
  - apiVersion: acme.cert-manager.io/v1
    blockOwnerDeletion: true
    controller: true
    kind: Order
    name: demo2-hsawant-tzlw7-616897090
    uid: 0263e448-de52-4624-b3ae-2cc5d1202daa
  resourceVersion: "489054492"
  uid: fe5714f0-57e1-45cc-8cd7-d2692bec06df
spec:
  authorizationURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/166452861706
  dnsName: demo2-hsawant.services-internal.dev.lacework.engineering
  issuerRef:
    group: cert-manager.io
    kind: ClusterIssuer
    name: letsencrypt
  key: zmIBcgg4GiXq4VB-BQz305XXOQBlD8J844bNE0JAEbI
  solver:
    dns01:
      route53:
        region: us-west-2
        secretAccessKeySecretRef:
          name: ""
  token: O0K9VUfttpqv4vFMieKvdMujKw71m3NPpWR7Zs9ai7Q
  type: DNS-01
  url: https://acme-v02.api.letsencrypt.org/acme/chall-v3/166452861706/Mp6z0g
  wildcard: false
status:
  presented: false
  processing: true
  state: pending
```

***kube-state-metrics Deployment***
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: victoria-metrics-kube-state-metrics
  namespace: default
  labels:    
    helm.sh/chart: kube-state-metrics-4.13.0
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/component: metrics
    app.kubernetes.io/part-of: kube-state-metrics
    app.kubernetes.io/name: kube-state-metrics
    app.kubernetes.io/instance: victoria-metrics
    app.kubernetes.io/version: "2.5.0"
    lacework.net/service_owner: sre
spec:
  selector:
    matchLabels:      
      app.kubernetes.io/name: kube-state-metrics
      app.kubernetes.io/instance: victoria-metrics
  replicas: 1
  template:
    metadata:
      labels:        
        helm.sh/chart: kube-state-metrics-4.13.0
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/component: metrics
        app.kubernetes.io/part-of: kube-state-metrics
        app.kubernetes.io/name: kube-state-metrics
        app.kubernetes.io/instance: victoria-metrics
        app.kubernetes.io/version: "2.5.0"
        lacework.net/service_owner: sre
    spec:
      hostNetwork: false
      serviceAccountName: victoria-metrics-kube-state-metrics
      securityContext:
        fsGroup: 65534
        runAsGroup: 65534
        runAsUser: 65534
      containers:
      - name: kube-state-metrics
        args:
        - --custom-resource-state-config
        - |
          spec:
            resources:
              - groupVersionKind:
                  group: acme.cert-manager.io
                  version: "v1"
                  kind: Challenge
                metrics:
                  - name: lacework_cert_manager_challenge_timestamp_seconds
                    help: "CertManager Challenge creation time in Unix Epoch format"
                    each:
                      type: Gauge
                      gauge:
                        path: [metadata, creationTimestamp]
                    labelsFromPath:
                      name: [metadata, name]
                      dnsName: [spec, dnsName]
                      type: [spec, type]
        - --port=8080
        - --resources=certificatesigningrequests,configmaps,cronjobs,daemonsets,deployments,endpoints,horizontalpodautoscalers,ingresses,jobs,limitranges,mutatingwebhookconfigurations,namespaces,networkpolicies,nodes,persistentvolumeclaims,persistentvolumes,poddisruptionbudgets,pods,replicasets,replicationcontrollers,resourcequotas,secrets,services,statefulsets,storageclasses,validatingwebhookconfigurations,volumeattachments,leases,challenges
        - --metric-labels-allowlist=pods=[lacework.net/service_owner],nodes=[lw-role,node.kubernetes.io/instance-type,spotinst.io/node-lifecycle,topology.kubernetes.io/region,topology.kubernetes.io/zone,kubernetes.io/arch],services=[lacework.net/service_owner],deployments=[lacework.net/service_owner],jobs=[lacework.net/service_owner],horizontalpodautoscalers=[lacework.net/service_owner]
        - --telemetry-port=8081
        imagePullPolicy: IfNotPresent
        image: "registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.5.0"
        ports:
        - containerPort: 8080
          name: "http"
        - containerPort: 8081
          name: "metrics"
        livenessProbe:
          httpGet:
            path: /healthz
            port: 8080
          initialDelaySeconds: 5
          timeoutSeconds: 5
        readinessProbe:
          httpGet:
            path: /
            port: 8080
          initialDelaySeconds: 5
          timeoutSeconds: 5
      imagePullSecrets:
        - name: some-docker-secret
```

***ERROR in kube-state-metrics logs***
```
E1019 21:21:59.062191       1 registry_factory.go:469] "kube_acme_cert-manager_io_v1_Challenge_lacework_cert_manager_challenge_timestamp_seconds" err="[]: [kind]: []: strconv.ParseFloat: parsing \"Challenge\": invalid syntax"
E1019 21:21:59.062218       1 registry_factory.go:469] "kube_acme_cert-manager_io_v1_Challenge_lacework_cert_manager_challenge_timestamp_seconds" err="[]: [metadata]: []: expected number but was map[creationTimestamp:2022-10-19T21:21:58Z finalizers:[finalizer.acme.cert-manager.io] generation:1 managedFields:[map[apiVersion:acme.cert-manager.io/v1 fieldsType:FieldsV1 fieldsV1:map[f:metadata:map[f:ownerReferences:map[.:map[] k:{\"uid\":\"0263e448-de52-4624-b3ae-2cc5d1202daa\"}:map[.:map[] f:apiVersion:map[] f:blockOwnerDeletion:map[] f:controller:map[] f:kind:map[] f:name:map[] f:uid:map[]]]] f:spec:map[.:map[] f:authorizationURL:map[] f:dnsName:map[] f:issuerRef:map[.:map[] f:group:map[] f:kind:map[] f:name:map[]] f:key:map[] f:solver:map[.:map[] f:dns01:map[.:map[] f:route53:map[.:map[] f:region:map[] f:secretAccessKeySecretRef:map[.:map[] f:name:map[]]]]] f:token:map[] f:type:map[] f:url:map[] f:wildcard:map[]]] manager:cert-manager-orders operation:Update time:2022-10-19T21:21:58Z] map[apiVersion:acme.cert-manager.io/v1 fieldsType:FieldsV1 fieldsV1:map[f:metadata:map[f:finalizers:map[.:map[] v:\"finalizer.acme.cert-manager.io\":map[]]] f:status:map[.:map[] f:presented:map[] f:processing:map[] f:state:map[]]] manager:cert-manager-challenges operation:Update time:2022-10-19T21:21:59Z]] name:demo2-hsawant-tzlw7-616897090-806614722 namespace:default ownerReferences:[map[apiVersion:acme.cert-manager.io/v1 blockOwnerDeletion:true controller:true kind:Order name:demo2-hsawant-tzlw7-616897090 uid:0263e448-de52-4624-b3ae-2cc5d1202daa]] resourceVersion:489054492 uid:fe5714f0-57e1-45cc-8cd7-d2692bec06df]"
E1019 21:21:59.062238       1 registry_factory.go:469] "kube_acme_cert-manager_io_v1_Challenge_lacework_cert_manager_challenge_timestamp_seconds" err="[]: [spec]: []: expected number but was map[authorizationURL:https://acme-v02.api.letsencrypt.org/acme/authz-v3/166452861706 dnsName:demo2-hsawant.services-internal.dev.lacework.engineering issuerRef:map[group:cert-manager.io kind:ClusterIssuer name:letsencrypt] key:zmIBcgg4GiXq4VB-BQz305XXOQBlD8J844bNE0JAEbI solver:map[dns01:map[route53:map[region:us-west-2 secretAccessKeySecretRef:map[name:]]]] token:O0K9VUfttpqv4vFMieKvdMujKw71m3NPpWR7Zs9ai7Q type:DNS-01 url:https://acme-v02.api.letsencrypt.org/acme/chall-v3/166452861706/Mp6z0g wildcard:false]"
E1019 21:21:59.062258       1 registry_factory.go:469] "kube_acme_cert-manager_io_v1_Challenge_lacework_cert_manager_challenge_timestamp_seconds" err="[]: [status]: []: expected number but was map[presented:false processing:true state:pending]"

```

**What you expected to happen**:
I expected Challenge CRD to be parsed and metrics created.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- kube-state-metrics version: 4.13.*

**Comment by olivierlemasle** [comment]
This seems to be fixed in kube-state-metrics 2.6.0.

Indeed, I was not able to reproduce the issue with `registry.k8s.io/kube-state-metrics/kube-state-metrics:v2.6.0`:

```
# HELP kube_acme_cert_manager_io_v1_Challenge_lacework_cert_manager_challenge_timestamp_seconds CertManager Challenge creation time in Unix Epoch format
# TYPE kube_acme_cert_manager_io_v1_Challenge_lacework_cert_manager_challenge_timestamp_seconds gauge
kube_acme_cert_manager_io_v1_Challenge_lacework_cert_manager_challenge_timestamp_seconds{dnsName="test.example.net",name="example-com2-942rq-2202368468-3401894249",type="HTTP-01"} 1.666635962e+09
```

 but I reproduced it with the `v2.5.0` image.

Please try to upgrade kube-state-metrics.

**Comment by olivierlemasle** [comment]
Actually, the fix was https://github.com/kubernetes/kube-state-metrics/pull/1754.
The error came from the `-` in the API group `acme.cert-manager.io`.
